src/rtk/features/singleRights/singleRightsSlice.ts

```diff
@@ -185,47 +194,92 @@ const singleRightSlice = createSlice({
   },
   extraReducers: (builder) => {
     builder
+      .addCase(delegationAccessCheck.pending, (state, action) => {
+        const serviceID = action.meta.arg.serviceResource.identifier;
+        let nextServicesWithStatus;
```

Comment:
Hva er det nextServicesWithStatus brukes til?

Reply:
Denne og de neste fire kommentarene handler vel egentlig om det samme, så jeg svarer bare på alle her:

Dette handler om state-handling, og jeg har prøvd å følge [React sine guidelines](https://react.dev/learn/updating-arrays-in-state) på hvordan man skal håndtere ending av lister i state.

Et recap:
![image](https://github.com/Altinn/altinn-access-management-frontend/assets/22012997/73797a46-ecb6-4e29-ad29-985e135aec6b)

Spesifikt har jeg her fulgt prinsippene rundt [transformering av arrays](https://react.dev/learn/updating-arrays-in-state#transforming-an-array), som anbefaler at man lager et nytt array (eksemplene til react bruker navnet 'next' etterfulgt av navnet på staten som skal oppdateres, så jeg har gjort det samme) som inneholder de nye endringene og så setter staten til dette nye arrayet.
`nextServicesWithStatus` er altså dette nye arrayet som kommer til å erstatte arrayet i staten `servicesWithStatus`.

Nå er jeg forsåvidt enig i at `nextServicesWithStatus` er et ganske langt variabelnavn, og `servicesWithStatus` er ikke akkurat mitt favorittnavn til å begynne med heller, så kanskje det er lettere hvis vi kaller den nye state-variablen `nextStateArray`? Tror du det ville hjulpet på forståelsen?

Reply:
Personlig synes jeg 'changed' foran er et bedre navn på dette enn 'next' som også gjør at andre som ikke har lest den react-docsen der kan skjønne hva propen er til for. Med det sagt så er nextStateArray bedre enn nextServicesWithStatus for da får man mer frem intensjonen med objektet

Reply:
Da endrer jeg til `nextStateArray` 👍

Reply:
Det absolutt beste tror jeg er nextStateArray med typen spesifisert, da får man frem begge deler

```suggestion
        let nextServicesWithStatus: ServiceWithStatus
```